pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,988 @@
+            - noise.sky :  (time-series)
+            - tfreq : transmission frequency (time-series)
+            - nave : number of averages  (time-series)
+            - cp : control program ID (time-series)
+            - p_l : Signal to Noise ratio (range-time)
+            - v : velocity (range-time)
+            - w_l : spectral width (range-time)
+            - elv : elevation (optional) (range-time)
+
+        Future Work
+        ------------
+        day-night terminators
+        slant ranges
+
+        Parameters
+        ----------
+        dmap_data: List[dict]
+            List of dictionaries of the data to be plotted containing the
+            parameter fields used in the summary plot.
+        beam_num : int
+        ax: matplotlib.axes
+            axes object for another way of plotting
+            Default: None
+        groundscatter : boolean
+            Flag to indicate if groundscatter should be plotted.
+            Placed only on the velocity plot.
+            Default : False
+        channel : int
+            channel number that will be plotted
+            in the summary plot.
+            Default: 'all'
+        figsize : (int,int)
+            tuple containing (height, width) figure size
+            Default: 11 x 8.5
+        cmaps: dict
+            dictionary of matplotlib color maps for the summary
+            plot parameters.
+            https://matplotlib.org/tutorials/colors/colormaps.html
+            Default: viridis
+            note: to reverse the color just add _r to the string name
+        boundary: dict
+            dict of the parameter names of a summary plot as key names:
+                - 'search.noise': search noise
+                - 'sky.noise': sky noise
+                - 'nave': number of averages
+                - 'tfreq': transmission frequency
+                - 'p_l': Signal to Noise
+                - 'v': velocity
+                - 'w_l': spectral width
+                - 'elv': elevation
+            with a tuple as the value (zmin, zmax) for the plots
+            Default: {'noise.search': (1e0, 1e5),
+                      'noise.sky': (1e0, 1e5),
+                      'tfreq': (8, 22),
+                      'nave': (0, 60),
+                      'p_l': (0, 30),
+                      'v': (-200, 200),
+                      'w_l': (0, 150),
+                      'elv': (0, 50)}
+        plot_elv: boolean
+            boolean determines if elevation should be plotted or not.
+            If there is no data for elevation data field then elevation is not
+            plotted.
+            Default: True
+        title: str
+            title of the plot
+            Default: auto-generated by the files details
+            {radar name} Fitacf {version} {start_date} - {end_date}  Beam {num}
+        Raises
+        ------
+        IndexError
+        RTPUnknownParameterError
+        RTPIncorrectPlotMethodError
+        RTPNoDataFoundError
```

Comment:
This section seems unnecessary, but whatever.  ;-) . It does need a white line separating it from the input parameters.

Reply:
Currently writing it with the numpy style guide, but I follow the numpy docstring style and prefer to include Raises because it allows non-code diving users to at least know what potential exceptions they may receive.

Fixed white space and added to the missing parameters in the above section.